include/SColor.h

```diff
@@ -416,6 +417,22 @@ namespace video
 			const float inv = 1.0f / 255.0f;
 			*this *= inv;
 		}
+		
+		inline static SColorf fromSRGB(SColorf&& input)
+		{
+			float color[3] = {input.r, input.g, input.b};
+			impl::SRGB2lin<float>(color);

```

Comment:
you could just pass `reinterpret_cast<float[3]>(input.pointer)` or `reinterpret_cast<float[3]>(&input.r)` instead

Reply:
then an `std::move()` of the `input` into the `SColorf` constructor, like this
```cpp
return SColorf(std::move(input));
```

Reply:
```impl::lin2SRGB<float>(reinterpret_cast<float[3]>(input.pointer));```
```impl::lin2SRGB<float>(reinterpret_cast<float[3]>(&input.r));```

Both failed on me (both MSVC and GCC): ```error: invalid cast from type ‘float*’ to type ‘float [3]’```

Reply:
ah, I forgot what the syntax for this is
https://stackoverflow.com/questions/17258749/is-there-a-way-of-casting-a-pointer-to-an-array-type